sql/src/main/java/io/crate/metadata/SearchPath.java

```diff
@@ -34,7 +36,7 @@
  * As writing fully qualified table names is usually tedious. This class models a list of schemas the system will use in
  * order to determine which table is meant by the user.
  */
-public final class SearchPath implements Iterable<String> {
+public final class SearchPath implements Collection<String> {
```

Comment:
do we really need to implement all these methods?
why not having a collection of Strings (composition) and implement only what we need, instead of implementing one?